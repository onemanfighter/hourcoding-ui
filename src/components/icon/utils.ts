import { ICON_LARGE_SIZE, ICON_MEDIUM_SIZE, ICON_SMALL_SIZE } from './constants';
import { IconProps } from './types';

const getIconSize = (size: IconProps['size']) => {
  return size
    ? size === 'small'
      ? ICON_SMALL_SIZE
      : size === 'medium'
        ? ICON_MEDIUM_SIZE
        : ICON_LARGE_SIZE
    : ICON_MEDIUM_SIZE;
};

export { getIconSize };
