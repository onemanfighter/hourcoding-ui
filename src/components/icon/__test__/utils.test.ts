import { getIconSize } from '../utils';

describe('Utils', () => {
  it('should return the 24 for small size', () => {
    const iconSize = getIconSize('small');

    expect(iconSize).toMatchSnapshot();
    expect(iconSize).toBe(24);
  });

  it('should return the 32 for medium size', () => {
    const iconSize = getIconSize('medium');

    expect(iconSize).toMatchSnapshot();
    expect(iconSize).toBe(32);
  });

  it('should return the 48 for large size', () => {
    const iconSize = getIconSize('large');

    expect(iconSize).toMatchSnapshot();
    expect(iconSize).toBe(48);
  });
});
