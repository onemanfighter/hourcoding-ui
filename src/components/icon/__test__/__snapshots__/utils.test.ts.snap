// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utils should return the 24 for small size 1`] = `24`;

exports[`Utils should return the 32 for medium size 1`] = `32`;

exports[`Utils should return the 48 for large size 1`] = `48`;
